aptos-move/aptos-vm/src/verifier/transaction_arg_validation.rs

```diff
@@ -41,10 +96,11 @@ static ALLOWED_STRUCTS: Lazy<BTreeMap<String, Option<ValidateArg>>> = Lazy::new(
 ///
 /// after validation, add senders and non-signer arguments to generate the final args
 pub(crate) fn validate_combine_signer_and_txn_args<S: MoveResolverExt>(
-    session: &SessionExt<S>,
+    session: &mut SessionExt<S>,
     senders: Vec<AccountAddress>,
     args: Vec<Vec<u8>>,
     func: &LoadedFunctionInstantiation,
+    struct_constructors_feature: bool,
```

Comment:
nit: Let's name this are_new_struct_constructors_enabled

Reply:
Done